src/pilot/deploy-overcloud.py

```diff
@@ -30,18 +36,534 @@
+                if "dell-compute" in server.name:
+                    dell_compute_nodes.append(((str(server_obj.ips(server)
+                                                ['ctlplane'][0]['addr']),
+                                                server.id)))
+
+            return dell_compute_nodes
+        except Exception as error:
+            message = "Exception {}: {}".\
+                      format(type(error).__name__, str(error))
+            logger.error("{}".format(message))
+            raise Exception("Failed to get the list of dell_compute nodes.")
+
+    def get_dell_compute_nodes_uuid(self):
+        try:
+            logger.info("Getting dell compute nodes uuid")
+            server_obj = servers.ServerManager(self.nova)
+            logger.info("Getting list of all dell compute nodes")
+            dell_compute_nodes_uuid = []
+            for server in server_obj.list():
+                if "dell-compute" in server.name:
+                    dell_compute_nodes_uuid.append(server.id)
+
+            return dell_compute_nodes_uuid
+        except Exception as error:
+            message = "Exception {}: {}".\
+                      format(type(error).__name__, str(error))
+            logger.error("{}".format(message))
+            raise Exception("Failed to get " +
+                            "the list of dell_compute nodes uuid.")
+
+
+class NodeConfig:
+    """
+    Description: Class responsible for node configuration operations
+    """
+
+    def __init__(self, ip_address):
+        self.ip_address = ip_address
+
+    def ssh_connect(self):
+        try:
+            logger.info("Establishing ssh connection with {}".format(self.ip_address))
+            # Initializing paramiko
+            ssh_conn = paramiko.SSHClient()
+            ssh_conn.set_missing_host_key_policy(paramiko.AutoAddPolicy())
+
+            # SSH into the remote host
+            ssh_conn.connect(self.ip_address, username='heat-admin')
+
+            # Save 'ssh_conn' in 'ssh_connection' class variable
+            self.ssh_connection = ssh_conn
+
+            logger.info("SSH connection to remote machine {} "
+                        "was successful".format(self.ip_address))
+        except Exception as error:
+            message = "Exception {}: {}". \
+                      format(type(error).__name__, str(error))
+            logger.error("Failed to establish SSH connection to remote machine {} "
+                         "with error {}".format(self.ip_address, message))
+
+    def close_ssh_connection(self):
+        self.ssh_connection.close()
+
+    def execute_cmd(self, command):
+        command = "sudo " + command
+        logger.info("Executing command {} on {}".format(command, self.ip_address))
+        time.sleep(1)
+        ssh_stdin, ssh_stdout, ssh_stderr = self.ssh_connection.cmd(command)
+        time.sleep(1)
+        return ssh_stdout.readlines(), ssh_stderr.readlines()
+
+    def update_filter(self):
```

Comment:
I would really prefer you change this to use the method of setting NovaSchedulerDefaultFilters in dell-environment.yaml.  As you point out in your writeup, it's simpler with fewer moving parts.

Reply:
Yeah, it's planned already. However, it's likely to happen after this PR.

Reply:
Upon reflection and after some checks, it turns out it was easier to switch to the the TripleO params method. Update_filter is now useless, hence removed. Edit_dell_environment_file is now handling those params.